homeassistant/components/rki_covid/sensor.py

```diff
@@ -0,0 +1,207 @@
+
+    try:
+        district = config_entry.data[ATTR_COUNTY]
+        sensors = [
+            RKICovidNumbersSensor(coordinator, district, info_type)
+            for info_type in SENSORS
+        ]
+        async_add_entities(sensors, update_before_add=True)
+    except KeyError:
+        _LOGGER.error("Could not determine %s from config!", ATTR_COUNTY)
+
+
+class RKICovidNumbersSensor(CoordinatorEntity):
+    """Representation of a sensor."""
+
+    name = None
+    unique_id = None
+
+    def __init__(
+        self,
+        coordinator: update_coordinator.DataUpdateCoordinator,
+        district: Dict[str, str],
+        info_type: str,
+    ):
+        """Initialize a new sensor."""
+        _LOGGER.debug(f"initialize {info_type} sensor for {district}")
+        super().__init__(coordinator)
+
+        data = coordinator.data[district]
+
+        if data.county:
+            self.name = f"{data.county} {info_type}"
+        else:
+            self.name = f"{data.name} {info_type}"
+        self.unique_id = f"{district}-{info_type}"
+        self.district = district
+        self.info_type = info_type
+        self.updated = datetime.now()
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return (
+            self.coordinator.last_update_success
+            and self.district in self.coordinator.data
+        )
+
+    @property
+    def state(self) -> Optional[str]:
+        """Return current state."""
+        try:
+            return getattr(self.coordinator.data[self.district], self.info_type)
+        except AttributeError:
+            return None
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return SENSORS[self.info_type]
+
+    @property
+    def state_class(self):
+        """Opt-in for long-term statistics."""
+        return STATE_CLASS_MEASUREMENT
+
+    @property
+    def unit_of_measurement(self):
+        """Return unit of measurement."""
+        if (
+            self.info_type == "count"
+            or self.info_type == "deaths"
+            or self.info_type == "recovered"
+        ):
+            return "people"
+        elif (
+            self.info_type == "weekIncidence"
+            or self.info_type == "hospitalizationIncidenceBaby"
+            or self.info_type == "hospitalizationIncidenceChildren"
+            or self.info_type == "hospitalizationIncidenceTeen"
+            or self.info_type == "hospitalizationIncidenceGrown"
+            or self.info_type == "hospitalizationIncidenceSenior"
+            or self.info_type == "hospitalizationIncidenceOld"
+        ):
+            return "nb"
+        else:
+            return "cases"
+
+    @property
+    def extra_state_attributes(self):
+        """Return device attributes."""
+        return {
+            ATTR_ATTRIBUTION: f"last updated {self.updated.strftime('%d %b, %Y  %H:%M:%S')} \n{ATTRIBUTION}"
+        }
```

Comment:
```suggestion
```
Attribution should be set with `_attr_attribution` and not be in `extra_state_attributes`

Reply:
For platforms it is necessary to use the extra_state_attributes as described here:
https://developers.home-assistant.io/docs/dev_101_states?_highlight=extra_state_attributes#using-states-in-your-component

Reply:
Atttribution should be your test that the data is collected from RKI
I suppose this `self.updated` contains a different time than when the sensor updated and that's why you want it as extra attributes?
Use another key in the dict and just have the datetime as the value in utc time and isoformatted

Reply:
I agree, changed to `_attr_attribution` during sensor initialization